member/views.py

```diff
@@ -143,34 +151,57 @@ def get_context_data(self, **kwargs):
-    bk = request.POST.get('bk', None)
-    band = get_object_or_404(Band, pk=bk)
-    emails = request.POST.get('emails', '').split('\n')
-
-    user_is_band_admin = Assoc.objects.filter(
-        member=request.user, band=band, is_admin=True).count() == 1
-
-    if not (user_is_band_admin or request.user.is_superuser):
-        raise PermissionDenied
+class InviteView(LoginRequiredMixin, FormView):
+    template_name = 'member/band_invite.html'
+    form_class = InviteForm
 
-    invited, in_band, invalid = [], [], []
-    for email in emails:
-        try:
-            validate_email(email)
-        except ValidationError:
-            invalid.append(email)
-            continue
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        band = get_object_or_404(Band, pk=self.kwargs['bk'])
+        context['band'] = band
+        return context
 
-        if Assoc.objects.filter(member__email=email, band=band).count() > 0:
-            in_band.append(email)
-        else:
-            Invite.objects.create(band=band, email=email, language=request.user.preferences.language)
-            invited.append(email)
+    def form_valid(self, form):
+        band = get_object_or_404(Band, pk=self.kwargs['bk'])
+        emails = form.cleaned_data['emails'].replace(',', ' ').split()
+
+        user_is_band_admin = Assoc.objects.filter(
+            member=self.request.user, band=band, is_admin=True).count() == 1
+
+        if not (user_is_band_admin or self.request.user.is_superuser):
+            raise PermissionDenied
+
+        invited, in_band, invalid = [], [], []
+        for email in emails:
+            try:
+                validate_email(email)
+            except ValidationError:
+                invalid.append(email)
+                continue
+
+            if Assoc.objects.filter(member__email=email, band=band).count() > 0:
+                in_band.append(email)
+            else:
+                Invite.objects.create(band=band, email=email, language=self.request.user.preferences.language)
+                invited.append(email)
+
+        if invalid:
+            messages.error(self.request,
+                           format_lazy(_('Invalid email addresses: {e}'),
+                                       e=join_trans(_(', '), invalid)))
+        if invited:
+            messages.success(self.request,
+                             format_lazy(_('Invitations sent to {e}'),
+                                         e=join_trans(_(', '), invited)))
+        if in_band:
+            messages.info(self.request,
+                          format_lazy(_('These users are already in the band: {e}'),
+                                      e=join_trans(_(', '), in_band)))
+
+        return super().form_valid(form)
 
-    return JsonResponse({'invited': invited, 'in_band': in_band, 'invalid': invalid})
+    def get_success_url(self):
+        return reverse('member-invite', args=[self.kwargs['bk']])
```

Comment:
If all of the emails are valid, we should return to the band page, not the invite page